src/utils/MastodonUtils.C

```diff
@@ -68,3 +71,125 @@ MastodonUtils::regularize(const std::vector<Real> & history_acc,
   }
   return {reg_tme, reg_acc};
 }
+
+bool
+MastodonUtils::checkVectorsUnequalSize(std::vector<std::vector<Real>> & vectors)
+{
+  for (auto v : vectors)
+  {
+    if (v.size() != vectors[0].size())
+      return true;
+  }
+  return false;
+}
+
+bool
+MastodonUtils::areVectorsEqual(std::vector<Real> & vector1,
+                               std::vector<Real> & vector2,
+                               Real percent_error)
+{
+  if (vector1.size() != vector2.size())
+    return false;
+  for (std::size_t i = 0; i < vector1.size(); ++i)
+  {
+    if (!MooseUtils::absoluteFuzzyEqual(
+            vector1[i], vector2[i], std::abs(vector1[i] * percent_error / 100)))
+      return false;
+  }
+  return true;
+}
+
+bool
+MastodonUtils::isVectorNonPositive(const std::vector<Real> & vector)
+{
+  for (auto element : vector)
+    if (element <= 0)
+      return true;
+  return false;
+}
+
+Real
+MastodonUtils::mean(std::vector<Real> & vector)
+{
+  Real sum = std::accumulate(vector.begin(), vector.end(), 0.0);
+  return sum / vector.size();
+}
+
+Real
+MastodonUtils::median(std::vector<Real> & vector, const std::string & interpolation)
```

Comment:
inputs should all be const

Reply:
I'm doing a sort on the vector in this particular function, so I'm not using const here. I changed the others..

Reply:
I would change this to const, and create a copy internally that is sorted. It doesn't seem correct that inorder to compute the median my vector will become sorted.